### PR TITLE
Configure dependabot

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

Initial [configuration of dependabot](https://dependabot.com/docs/config-file/) which covers JS, Go, and Docker. Interested in seeing what we can do with this.

### CODE REVIEW

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - tinyels
+      - chrisgilmerproj
+      - jacquelineIO
+      - sarboc
+      - chrisrcoles
+
+  # Keep Dockerfile up to date, batching pull requests weekly
+  - package_manager: "docker"
+    directory: "/"
+    update_schedule: "weekly"
```
Comment (mkrump) [inline .dependabot/config.yml]: why not run daily? looks like it limits to weekdays only https://dependabot.com/docs/config-file/#update_schedule-required.

File: .dependabot/config.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - chrisgilmerproj
+      - jacquelineIO
+      - sarboc
+      - chrisrcoles
+
+  # Keep go.mod (& go.sum) up to date as soon as
+  # new versions are published
+  - package_manager: "go:modules"
+    directory: "/"
+    update_schedule: "weekly"
```
Comment (mkrump) [inline .dependabot/config.yml]: why not run daily? looks like it limits to weekdays only https://dependabot.com/docs/config-file/#update_schedule-required.

File: .dependabot/config.yml (hunk 1 above)
Comment (chrisgilmerproj) [reply .dependabot/config.yml]: I was worried about overloading folks as we turn this on. Also the ATO only requires we do a weekly check. But its probably something we'll bump up as we learn how it behaves.

File: .dependabot/config.yml (hunk 2 above)
Comment (chrisgilmerproj) [reply .dependabot/config.yml]: Same as above. I only turned JS onto daily right now.

File: .dependabot/config.yml (hunk 1 above)
Comment (mkrump) [reply .dependabot/config.yml]: Yeah it limits the total number of PRs that can be opened at one time, so will just catch up faster w/ daily updates.

Comment (chrisgilmerproj) [reply .dependabot/config.yml]: Alright, I'll bump these.

**Comment by mkrump** [comment]
LTGM. One question though do you think it's worth trying to set the automerge conditions here? https://dependabot.com/docs/config-file/#automerged_updates I suspect branch protection approval condition will still block us, but this seems cleaner and it allows you to say only automerge patches for instance.

**Comment by chrisgilmerproj** [comment]
> LTGM. One question though do you think it's worth trying to set the automerge conditions here? https://dependabot.com/docs/config-file/#automerged_updates I suspect branch protection approval condition will still block us, but this seems cleaner and it allows you to say only automerge patches for instance.

I was going to try out the automerge in a different PR :) I absolutely think we ought to try out automerge though and possibly remove some of our other tooling if it works.

**Pull request merged**